autoextract/aio/errors.py

```diff
@@ -1,11 +1,52 @@
 # -*- coding: utf-8 -*-
+import logging
+import re
+from typing import Optional
+
 from aiohttp import ClientResponseError
+from tenacity import RetryError
+
+logger = logging.getLogger(__name__)
+
+
+class DomainOccupied:
+
+    DOMAIN_OCCUPIED_REGEX = re.compile(
+        r".*domain (.+) is occupied, please retry in (.+) seconds.*",
+        re.IGNORECASE
+    )
+
+    DEFAULT_RETRY_SECONDS = 5 * 60  # 5 minutes
 
+    def __init__(self, domain: str, retry_seconds: float):
+        self.domain = domain
+        self.retry_seconds = retry_seconds
 
-class ApiError(ClientResponseError):
-    """ Exception which is raised when API returns an error.
+    @classmethod
+    def from_message(cls, message: str) -> Optional["DomainOccupied"]:
+        match = cls.DOMAIN_OCCUPIED_REGEX.match(message)
+        if not match:
+            return None
+
+        domain = match.group(1)
+
+        try:
+            retry_seconds = float(match.group(2))
+        except ValueError:
+            logger.warning(
```

Comment:
I always have issues when deciding the proper log level.
Lately, I've been trying to use `warning` as something that we could act on.
Of course, if the regex is wrong, we'll get an warning, but on the main case, should it be a warning or simply info/debug?

Reply:
Perhaps it's closer to a warning because it would require:
- a fix on the API;
- or a fix in our retry seconds parser